Sources/MongoMobile/MongoMobile.swift

```diff
@@ -14,37 +14,43 @@ public struct MongoClientSettings {
     }
 }
 
+/// A MongoDB server error code.
+/// - SeeAlso: https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err
+public typealias ServerErrorCode = Int
+
 public enum MongoMobileError: LocalizedError {
-    case invalidClient()
-    case invalidInstance(message: String)
-    case invalidLibrary()
-    case instanceDropError(message: String)
-    case cleanupError(message: String)
+    /// Thrown when MongoMobile is incorrectly used.
+    case logicError(message: String)
+
+    /// Thrown when an embedded mongo server instance returns an error.
+    /// - SeeAlso: https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err
+    case embeddedMongoError(code: ServerErrorCode, message: String)
+
+    /// Thrown when MongoMobile encounters internal errors it cannot recover from.
+    case internalError(message: String)
 
     /// a string to be printed out when the error is thrown
     public var errorDescription: String? {
         switch self {
-        case let .invalidInstance(message),
-            let .instanceDropError(message),
-            let .cleanupError(message):
+        case let .embeddedMongoError(_, message),
+             let .internalError(message),
+             let .logicError(message):
             return message
-        default:
-            return nil
         }
     }
 }
 
-public struct MongoEmbeddedV1Error: LocalizedError {
-    private let statusMessage: String
-    private let error: mongo_embedded_v1_error
-
-    public var errorDescription: String? {
-        return "\(error): \(statusMessage)"
-    }
-
-    internal init(_ error: mongo_embedded_v1_error, statusMessage: String) {
-        self.statusMessage = statusMessage
-        self.error = error
+/// Given an `OpaquePointer` to a `mongo_embedded_v1_status`, get an appropriate `MongoMobileError`.
+internal func parseEmbeddedMongoError(_ status: OpaquePointer?) -> MongoMobileError {
```

Comment:
since this is internal anyway, should we rename to `parseEmbeddedMongoErrorV1`?

Reply:
On a side note, does it make sense to start making type aliases for these types to better document the code:
```
internal typealias mongo_embedded_v1_status_t OpaquePointer
internal func parseEmbeddedMongoErrorV1(_ status: mongo_embedded_v1_status_t?) -> MongoMobileError {
```

Reply:
also, I think this is maybe named backwards: `parse**MongoEmbedded**ErrorV1`

Reply:
Added your suggested typealias, refactored to `parseMongoEmbeddedV1Status` (since its actually parsing a status, not an error).

Reply:
do you think we should make those typealiases in general? In this case I think I accidentally added a `_t` to the name which isn't there in the official code, maybe that's a caveat of this approach

Reply:
I think for C types that we cannot import, typealiasing is worthwhile. It communicates context at the type level with no real cost. We do run the risk of typos like the one I just did, but when naming anything we run that risk.

Importing the type fully and using `SomePointer<type>` is definitely preferable, but I assume that isn't possible here for one reason or another.

Reply:
I generally agree with adding typealiases for more descriptive labeling purposes. 

I have somewhat mixed feelings on using them with `OpaquePointer`s though, because I think it kind of creates the illusion that we are using some strong type. any `OpaquePointer` can be successfully cast to a `mongo_embedded_v1_status`, but that doesn't mean it's actually a pointer to one and can be passed to `parseEmbeddedMongoErrorV1`. 
compare this to e.g. `ServerErrorCode` - any `Int` can be cast to a `ServerErrorCode`, *and* can be used as one with no issues. 
`OpaquePointer` is admittedly very vague, so I see the labeling advantage, but at least it doesn't hide the fact that we have no guarantee about the underlying type and need to use it carefully. 
I can imagine some future situation where we try to cast to one of these typealiases, or switch on one, and get unexpected results. maybe the compiler complains about that though.

I'm ok with this particular usage but I'm hesistant about adopting this pattern everywhere.

Reply:
We could define some wrapper like:
```swift
internal struct mongo_embedded_v1_status {
    let ptr: OpaquePointer?
}
```
which would then force a cast (well, initialization), making implicit bad behavior less possible. This would be similar to a golang `type` (which would actually be kind of nice here...).

Reply:
I think its fine as-is, let's not hold this up any longer